src/commands/MeetingNotes.ts

```diff
@@ -29,6 +34,68 @@ export default class MeetingNotes extends Command {
     }, definition);
   }
 
+  /**
+   * Shortens the given long link into acmurl.com/{shortlink}.
+   * @param interaction The command interaction from the called command, so we can edit it in the event of an error.
+   * @param shortlink The shortened ID to shorten the link to (acmurl.com/{shortlink})
+   * @param longlink The long link to shorten (here, a link to the Notion meeting notes)
+   * @returns The new shortened link, or null in the event of an error.
+   */
+  private async makeACMURL(
+    interaction: CommandInteraction, shortlink: string, longlink: string,
+  ): Promise<string | null> {
+    const linkTitle = `Discord Bot: ${shortlink}`;
+    const config: ACMURLConfig = {
+      username: this.client.settings.acmurl.username,
+      password: this.client.settings.acmurl.password,
+    };
+    try {
+      const shortURL = await addACMURL(shortlink, longlink, linkTitle, config);
+      return shortURL;
+    } catch (e) {
+      // We might error out if an ACMURL already exists with the provided shortlink.
+      // We'll attempt to handle that by updating the ACMURL.
+      const errorUUID: UUIDv4 = newUUID();
+
+      const error = e as any;
+
+      // If the error we get is specifically that a ACMURL already existed.
+      if (error.message === 'error:keyword') {
+        try {
+          // Make a new one and return the old long link and new ACMURL
+          const [, newURL] = await handleExistingACMURL(
+            shortlink, longlink, linkTitle, config,
+          );
+          return newURL;
+        } catch (e2) {
+          const updateError = e2 as any;
+          // If by any chance there's an error when updating the ACMURL, log it and return.
```

Comment:
This feels a little weird that we have to throw an error before we can figure out that if acmurl exists and hopefully separate it out from other types of errors.

Two possible options:
1. Can we confirm if the acmurl exists or not first and then conditionally decide if we have to handle an existing URL or not?
2. Can we just always treat it like the ACMURL exists and we're updating it and then check if the old longlink is undefined - which would mean there was no previous URL

Reply:
We'll fix this error when we update ACMURL to have a landing page/better update functionality.
This code was originally taken from BreadBot's /acmurl command, which tracks what the previous longlink was after shortening it, but there isn't a good way for us to go straight to updating it without catching the error because of how the routes are structured on the ACMURL server.